internal/encryption/method/aesgcm/README.md

```diff
@@ -0,0 +1,51 @@
+# AES-GCM encryption method
+
+This folder contains the state encryption implementation of the AES-GCM encryption method.
+
+This is implemented following the guidance of the following document: ([NIST SP 800-38D](https://csrc.nist.gov/pubs/sp/800/38/d/final)).
+
+
+## Configuration
+
+You can configure the encryption by specifying the following method block:
+
+```hcl2
+terraform {
+  encryption {
+    method "aes_gcm" "mymethod" {
+      # Pass the key provider with a 16, 24, or 32 byte encryption key here:
+      keys = key_provider.someprovider.somename
+      
+      # Leave the AAD empty unless needed. Pass as a list of bytes if needed:  
+      aad  = [1,2,3,4,...]
+    }
+  }
+}
+```
+
+| Field               | Description                                                                                                                                                                                      |
+|---------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `keys` (*required*) | Encryption and decryption key, as well as key provider metadata.                                                                                                                                 |

```

Comment:
Why is this plural "keys"? This is very confusing to me. Esp. since it's defined as singular "Encryption and decryption key".

Reply:
The key provider returns an object containing an `encryption_key` and a `decryption_key` field. There is more than one key involved. Alternatively, we can name it `key_provider`, but that's inaccurate on a technical level, the user could (if they were into causing themselves pain) pass a list of individual bytes as keys.